web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -19,58 +19,289 @@
+  const [password, setPassword] = useState(passwordProp || "");
+
+  useEffect(() => {
+    if (password.length === 0) onValidate(false);
+  }, [password, onValidate]);
+
+  const changePassword = (v) => setPassword(v);
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    onSubmit(password);
+  };
+
+  return (
+    <Form id={id} onSubmit={submitForm}>
+      <PasswordAndConfirmationInput
+        id="encryptionPasswordInput"
+        value={password}
+        onChange={changePassword}
+        onValidation={onValidate}
+      />
+    </Form>
+  );
+};
+
+const EncryptionPasswordField = ({ selected: selectedProp, password: passwordProp, onChange, isLoading }) => {
+  const [selected, setSelected] = useState(selectedProp || false);
+  const [password, setPassword] = useState(passwordProp || "");
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [isFormValid, setIsFormValid] = useState(true);
+
+  const openForm = () => setIsFormOpen(true);
+
+  const closeForm = () => setIsFormOpen(false);
+
+  const acceptForm = (newPassword) => {
+    closeForm();
+    setPassword(newPassword);
+    onChange({ selected, password: newPassword });
+  };
+
+  const cancelForm = () => {
+    closeForm();
+    if (password.length === 0) setSelected(false);
+  };
+
+  const validateForm = (valid) => setIsFormValid(valid);
+
+  const changeSelected = (value) => {
+    setSelected(value);
+
+    if (value && password.length === 0) openForm();
+
+    if (!value) {
+      setPassword("");
+      onChange({ selected: false, password: "" });
+    }
+  };
+
+  const ChangePasswordLink = () => {
+    return <Button variant="link" onClick={openForm}>Change password</Button>;
+  };
+
+  if (isLoading) return <Skeleton width="25%" />;
 
   return (
-    <Section title="Settings" openDialog={openSettings}>
-      <Popup title="Settings" isOpen={isOpen}>
-        <ProposalSettingsForm
-          id="settings-form"
-          proposal={proposal}
-          onSubmit={onSettingsChange}
-          onValidate={setIsValid}
+    <>
+      <div className="split">
+        <Switch
+          id="encryption"
+          label="Encrypt devices"
+          isReversed
+          isChecked={selected}
+          onChange={changeSelected}
+        />
+        { selected && <ChangePasswordLink /> }
+      </div>
+      <Popup aria-label="Change encryption password" title="Devices encryption" isOpen={isFormOpen}>
+        <EncryptionPasswordForm
+          id="encryptionPasswordForm"
+          password={password}
+          onSubmit={acceptForm}
+          onValidate={validateForm}
         />
         <Popup.Actions>
-          <Popup.Confirm form="settings-form" type="submit" isDisabled={!isValid}>Accept</Popup.Confirm>
-          <Popup.Cancel onClick={() => setIsOpen(false)} autoFocus />
+          <Popup.Confirm form="encryptionPasswordForm" type="submit" isDisabled={!isFormValid}>Accept</Popup.Confirm>
+          <Popup.Cancel onClick={cancelForm} autoFocus />
```

Comment:
Here is the problem with the autoFocus. We are forcing the dialog to put the focus in the Cancel action. Probably a copy/paste issue :)

Reply:
So, we should have a look to all Popup.Actions